src/classes/BusinessLogicServices/ShopOrderService.php

```diff
@@ -0,0 +1,335 @@
+        foreach ($sourceOrderItems as $sourceOrderItem) {
+            /** @var \ProductCore $product */
+            $product = new \Product((int)$sourceOrderItem['product_id']);
+            if (!$product->is_virtual) {
+                $orderItem = $this->getOrderItem($sourceOrderItem, $defaultParcel);
+
+                $orderItem->setPrice((float)$sourceOrderItem['unit_price_tax_excl']);
+                $orderItem->setTotalPrice((float)$sourceOrderItem['unit_price_tax_incl']);
+
+                $orderItems[] = $orderItem;
+            }
+        }
+
+        return $orderItems;
+    }
+
+    /**
+     * Sets additional order item information (title, quantity, category...).
+     *
+     * @param array $sourceOrderItem PrestaShop order item.
+     *
+     * @param \Packlink\BusinessLogic\Http\DTO\ParcelInfo $defaultParcel
+     *
+     * @return \Packlink\BusinessLogic\Order\Objects\Item
+     *
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    private function getOrderItem($sourceOrderItem, $defaultParcel)
+    {
+        $orderItem = new Item();
+        $product = new \Product((int)$sourceOrderItem['product_id']);
+        $languageId = (int)\Context::getContext()->language->id;
+
+        $orderItem->setQuantity((int)$sourceOrderItem['product_quantity']);
+
+        if (!empty($product->name)) {
+            $orderItem->setTitle($product->name[$languageId]);
+        }
+
+        $category = new \Category((int)$product->id_category_default);
+        if (!empty($category->name)) {
+            $orderItem->setCategoryName($category->name[$languageId]);
+        }
+
+        $orderItem->setWeight(round((float)$product->weight ?: $defaultParcel->weight, 2));
+        $orderItem->setWidth(ceil((float)$product->width ?: $defaultParcel->width));
+        $orderItem->setLength(ceil((float)$product->depth ?: $defaultParcel->length));
+        $orderItem->setHeight(ceil((float)$product->height ?: $defaultParcel->height));
+
+        /** @var array $productCoverImage */
+        $productCoverImage = \Image::getCover($product->id);
+        if (!empty($productCoverImage)) {
+            $link = new \Link();
+            $productImageUrl = $link->getImageLink(
+                $product->link_rewrite[$languageId],
+                (int)$productCoverImage['id_image'],
+                \ImageType::getFormatedName('home')
+            );
+            $orderItem->setPictureUrl($productImageUrl);
+        }
+
+        return $orderItem;
+    }
+
+    /**
+     * Gets the order.
+     *
+     * @param int $orderId Shop order ID.
+     *
+     * @return PrestaShopOrder
+     * @throws \Packlink\BusinessLogic\Order\Exceptions\OrderNotFound
+     */
+    private function getOrder($orderId)
+    {
+        $order = null;
+        try {
+            $order = new PrestaShopOrder($orderId);
+        } catch (\PrestaShopDatabaseException $e) {
+        } catch (\PrestaShopException $e) {
+        }
```

Comment:
Why do you use empty catch structure

Reply:
Because if an exception occurred, the validation will handle it on the next line.

Reply:
If exception occurred then order will be null you could throw exception into this clauses

Reply:
Then, we would need to check two cases, one for exception and one if the object is created but not loaded, which can happen in PrestaShop.

Reply:
Only need to check one , if exception was thrown you could throw the exception otherwise if exception was not throw you could check into try clause

Reply:
No need to throw the exception additionally, just loggin the validation is enouth.